src/openrct2/ride/Vehicle.cpp

```diff
@@ -9043,9 +9022,9 @@ void Vehicle::UpdateTrackMotionMiniGolfVehicle(Ride* curRide, rct_ride_entry* ri
     goto loc_6DCD2B;
 
 loc_6DCA9A:
-    regs.ax = track_progress - 1;
-    if (static_cast<uint16_t>(regs.ax) != 0xFFFF)
+    if (track_progress)
```

Comment:
```suggestion
    if (track_progress != 0)
```


ONLY use `if (var)` when handling a boolean.